### PR TITLE
Move indexer connector engine building

### LINKED ISSUE #25323: Installation of Wazuh Agent for collection of data from RTU/ PLC
Hi All, 
I am working on a project where I need to collect logs and events from Operational Technology devices (SCADA) like Remote Terminal Units (RTUs) and Programmable Logic Controllers (PLCs). 

The RTU & PLC units we have run a web server, and we parse the web pages (web scraping using Python), filter and extract relevant data, and use them. 

Is there a method by which Wazuh agents can be used to collect these? Has someone tried this out? 

Would love to hear from someone who has, and would help immensely to learn about the mechanism as well as their experiences. 

Thanks & Regards,

### LINKED ISSUE #25325: Move the indexer connector to the engine building process
## Description
This issue aims to move the indexer connector code and tests into the engine building process as a static library used by some components from the engine.

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|Closes #25325 |

This PR aims to move the indexer connector module into the engine build process.

### CODE REVIEW

**Review by sebasfalcone — COMMENTED** [review]
Is this directory structure right? (base -> include -> base)
```
├── base
│   ├── CMakeLists.txt
│   ├── include
│   │   └── base
│   │       ├── baseTypes.hpp
│   │       ├── dotPath.hpp
│   │       ├── error.hpp
│   │       ├── expression.hpp
│   │       ├── graph.hpp
│   │       ├── json.hpp
```

File: src/engine/README.md [hunk 1]
```diff
@@ -165,18 +165,21 @@ Documentation is generated with docxygen, currently html and latex outputs are g
...
-mkdir build
-cd build
-cmake ..
-make
+cd ..
+git clone https://github.com/microsoft/vcpkg.git
+cd vcpkg && ./bootstrap-vcpkg.sh
+export VCPKG_ROOT=$(pwd)
+export PATH=$VCPKG_ROOT:$PATH
+cd ../wazuh-http-request
```
Comment (sebasfalcone) [inline src/engine/README.md:175]: Wrong path

**Comment by Dwordcito** [comment]
> Is this directory structure right? (base -> include -> base)
> 
> ```
> ├── base
> │   ├── CMakeLists.txt
> │   ├── include
> │   │   └── base
> │   │       ├── baseTypes.hpp
> │   │       ├── dotPath.hpp
> │   │       ├── error.hpp
> │   │       ├── expression.hpp
> │   │       ├── graph.hpp
> │   │       ├── json.hpp
> ```

currently yes, because it is inherited from the engine, and we may have to change many things here to accommodate it.

**Review by sebasfalcone — CHANGES_REQUESTED** [review]

File: src/engine/source/indexerconnector/include/indexerConnector.hpp [hunk 2]
```diff
@@ -50,21 +50,13 @@ class EXPORTED IndexerConnector final
...
-                              const std::function<void(const int,
-                                                       const std::string&,
-                                                       const std::string&,
-                                                       const int,
-                                                       const std::string&,
-                                                       const std::string&,
-                                                       va_list)>& logFunction = {},
                               const uint32_t& timeout = DEFAULT_INTERVAL,
-                              const uint8_t workingThreads = 1);
+                              uint8_t workingThreads = 1);
```
Comment (sebasfalcone) [inline src/engine/source/indexerconnector/include/indexerConnector.hpp:59]: The const qualifiers was removed from here but not on the Cpp file. It was on purpose to remove it?

Comment (Dwordcito) [reply src/engine/source/indexerconnector/include/indexerConnector.hpp:59]: const-qualification of parameters only has an effect in function definitions

**Review by sebasfalcone — APPROVED** [review]

**Pull request merged**